File: Set1_1000/205_Isomorphic_Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        dict = {}
        
        for x, y in enumerate(s):
            if y not in dict:
                if t[x] in dict.values(): 
                    return False
                dict[y] = t[x]
            else:
                if dict[y] != t[x]:
                    return False
        return True
                

File: Set1_1000/205_Isomorphic_Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int wordIndex [] = new int [256];
        int patternIndex [] = new int [256];
        for (int i=0; i<Math.min(s.length(), t.length()); i++) {
            if (wordIndex[s.charAt(i)] != patternIndex[t.charAt(i)])
                return false;
            wordIndex[s.charAt(i)] = i + 1;
            patternIndex[t.charAt(i)] = i + 1;
        }
        return true;
    }
}
